src/citrine/resources/material_run.py

```diff
@@ -139,8 +139,14 @@ def get_history(self, id: Union[str, UUID, LinkByUID, MaterialRun]) -> MaterialR
             ]
         }
         data = self.session.post_resource(path, json=query)
-
-        return MaterialRun.build(data)
+        if data and data[0]["roots"]:
+            # Since the above query presents a single dataset to the endpoint, the response will be
+            # a list of length one, with a single route.
+            history_data = data[0]
+            history_data["roots"] = history_data["roots"][0]
+            return MaterialRun.build(history_data)
```

Comment:
This was a change that came from Pablo running the e2es.

The result of the endpoint changed slightly, such that it can now encode multiple material histories. He confirmed to me that because this call is querying on a single dataset ID, it will always return a single element, and its `roots` field will be an array of length 1.

I could have put this stuff in `_pre_build`, but my impression is we very well might want to expose the whole thing at some point in the future, so I'd make it clear these changes apply to this specific usage.

Reply:
Agreed, it's not great. And yeah, I have no problem making that tweak. Looking at the code in `GEMDResource`, it should work fine.